src/Dto/Segment/CreateSegment.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segment;
+
+use OneSignal\Dto\AbstractDto;
+use OneSignal\Dto\Filter\AbstractFilter;
+
+class CreateSegment implements AbstractDto
+{
+    /**
+     * @var non-empty-string|null
+     */
+    protected ?string $name = null;
+
+    /**
+     * @var array<int, AbstractFilter>
+     */
+    protected ?array $filters = null;
+
+    /**
+     * @param non-empty-string|null           $name
+     * @param array<int, AbstractFilter>|null $filters
+     */
+    public function __construct(string $name = null, array $filters = null)
+    {
+        $this->name = $name;
+        $this->filters = $filters;
+    }
+
+    /**
+     * @param non-empty-string $name
+     */
+    public function name(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * @param list<AbstractFilter> $filters
+     */
+    public function filters(array $filters): self
+    {
+        $this->filters = $filters;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return array_filter([
+            'name' => $this->name,
+            'filters' => array_map(
+                static function (AbstractFilter $filter): array {
+                    return $filter->toArray();
+                },
+                $this->filters ?? []
+            ),
```

Comment:
```suggestion
            'filters' => $this->filters !== null ? array_map(static function (AbstractFilter $filter): array {
                return $filter->toArray();
            }, $this->filters) : null,
```
So the `filters` won't be included if not set

Reply:
done